src/methods/tx/_events.ts

```diff
@@ -0,0 +1,345 @@
+import {
+  ChainBlockEventsCallback,
+  ExchangeTxEventData,
+  MergedPairedEvents,
+} from '../../types';
+import { getHdxEventEmitter } from '../../utils/eventEmitter';
+import BigNumber from 'bignumber.js';
+
+let mergedPairedEvents: MergedPairedEvents = {};
+
+const mergeEventToScope = (receivedEventData: any) => {
+  const intentionId = receivedEventData.data.id;
+  let pairedEventData = mergedPairedEvents[intentionId];
+
+  const getMergeStatus = () => {
+    const { status } = pairedEventData;
+
+    /**
+     * Status property can be changed only to true.
+     */
+    return {
+      ready: receivedEventData.status.ready || status!.ready,
+      inBlock: receivedEventData.status.inBlock || status!.inBlock,
+      finalized: receivedEventData.status.finalized || status!.finalized,
+      error: [...status!.error, ...receivedEventData.status.error],
+    };
+  };
+
+  if (pairedEventData === undefined) {
+    pairedEventData = receivedEventData;
+  } else {
+    pairedEventData = {
+      dispatchInfo: [
+        ...(pairedEventData.dispatchInfo || []),
+        ...(receivedEventData.dispatchInfo || []),
+      ],
+      section: [...pairedEventData.section, ...receivedEventData.section],
+      method: [...pairedEventData.method, ...receivedEventData.method],
+      status: getMergeStatus(),
+      data: {
+        ...pairedEventData.data,
+        ...receivedEventData.data,
+        directTrades: [
+          ...(pairedEventData.data.directTrades || []),
+          ...(receivedEventData.data.directTrades || []),
+        ],
+      },
+    };
+  }
+
+  /**
+   * Transaction values calculations
+   */
+
+  /**
+   * Calculate "match" value - total amount, which has been traded by Direct trade
+   */
+  if (
+    receivedEventData.method[0] === 'IntentionResolvedDirectTrade' &&
+    pairedEventData.data !== undefined &&
+    pairedEventData.data.directTrades !== undefined
+  ) {
+    let totalDirectTradeMatch = new BigNumber(0);
+    pairedEventData.data.directTrades.forEach(item => {
+      totalDirectTradeMatch = totalDirectTradeMatch.plus(item.amountReceived);
+    });
+    pairedEventData.data.match = totalDirectTradeMatch;
+  }
+
+  // TODO return final trade price and final trade fee
+
+  /**
+   * Calculate "totalAmountFinal" - total amount from all types of trading for
+   * this specific exchange action + fees.
+   */
+
+  const totalAmmTradeAmount: BigNumber =
```

Comment:
Is `amm` still viable? Just asking because you've removed all methods from amm to xyk

Reply:
Yeah, I need rename my own vars as well regarding last naming convention.